src/main/java/apoc/text/Strings.java

```diff
@@ -160,4 +160,12 @@ public String format(@Name("text") String text, @Name("params") List<Object> par
         if (params == null) return text;
         return String.format(Locale.ENGLISH,text, params.toArray());
     }
+    
+    @UserFunction
+    @Description("apoc.text.slug(text, delim) - slug the text with the given delimiter")
+    public String slug(@Name("text") String text, @Name(value = "delim", defaultValue = "-") String delim) {
+        if (text == null) return null;
+        if (delim == null) return null;
+        return text.replace(" ", delim).replace(delim+delim, delim);
```

Comment:
probably use replaceAll("\s+",delim) for all whitespace
or \W+ for non-word-characters